### PR TITLE
Enhancement - modified default thread pool settings to optimize Async…

### PR DESCRIPTION
**Pull request by kazoompa**
… and especially nested Async calls.

### CODE REVIEW

File: mica-core/src/main/java/org/obiba/mica/config/EventBusConfiguration.java [hunk 1]
```diff
@@ -27,11 +28,14 @@
 import com.google.common.eventbus.EventBus;
 import com.google.common.eventbus.Subscribe;
 
+import static org.bouncycastle.asn1.x500.style.RFC4519Style.name;
```
Comment (firone) [inline mica-core/src/main/java/org/obiba/mica/config/EventBusConfiguration.java]: Is it used ?

File: mica-core/src/main/java/org/obiba/mica/config/AsyncConfiguration.java [hunk 2]
```diff
@@ -52,15 +52,25 @@ public AsyncUncaughtExceptionHandler getAsyncUncaughtExceptionHandler() {
...
+
+  @Bean(name="opalExecutor")
+  public Executor getOpalAsyncExecutor() {
+    log.debug("Creating Async Task Executor");
+    ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+    executor.setCorePoolSize(propertyResolver.getProperty("opal.corePoolSize", Integer.class, DEFAULT_POOL_SIZE));
+    executor.setMaxPoolSize(propertyResolver.getProperty("opal.maxPoolSize", Integer.class, DEFAULT_MAX_POOL_SIZE));
+    executor.setQueueCapacity(propertyResolver.getProperty("queueCapacity", Integer.class, DEFAULT_QUEUE_CAPACITY));
+    executor.setThreadNamePrefix("mica-opal-executor-");
+    return new ExceptionHandlingAsyncTaskExecutor(executor);
```
Comment (firone) [inline mica-core/src/main/java/org/obiba/mica/config/AsyncConfiguration.java]: It's dangerous ... We wait that queue is full to add thread and if we add another task before another task is finished, there is an error ...
Executors.newFixedThreadPool() can be a better solution. We set a fixed number of thread that are already running, si no probleme of queue size.

Comment (firone) [inline mica-core/src/main/java/org/obiba/mica/config/AsyncConfiguration.java]: Or at least set maxPoolSize = corePoolSize with a high queue capacity

Comment (apruden) [inline mica-core/src/main/java/org/obiba/mica/config/AsyncConfiguration.java]: accepting this PR until a better executor strategy is tested. ideally should have bounded queue size and increase threads before the queue is full.

**Pull request merged**